R/select.R

```diff
@@ -0,0 +1,74 @@
+
+select_pos <- function(.x,
+                       ...,
+                       .include = NULL,
+                       .exclude = NULL,
+                       .strict = TRUE) {
+  vctrs::vec_assert(.x)
+
+  vars <- names(.x)
+  if (is_null(vars)) {
+    abort("Can't select within an unnamed vector.")
+  }
+
+  scoped_vars(vars)
+  select_impl(
+    .x,
+    ...,
+    .include = .include,
+    .exclude = .exclude,
+    .strict = .strict
+  )
+}
+
+# Example of implementation, mainly used for unit tests
+vec_select <- function(.x, ..., .strict = TRUE) {
+  if (is.data.frame(.x)) {
+    abort("`.x` can't be a data frame.")
+  }
+  idx <- select_pos(.x, ..., .strict = .strict)
+  set_names(vctrs::vec_slice(.x, idx), names(idx))
```

Comment:
Maybe just use `[` here? Given that it's only an example and that would make it work with data frames, which would be nice for testing?